packages/tizen_rpc_port/lib/src/stub_base.dart

```diff
@@ -0,0 +1,97 @@
+// You have generated a new plugin project without specifying the `--platforms`
+// flag. A plugin project with no platform support was generated. To add a
+// platform, run `flutter create -t plugin --platforms <platforms> .` under the
+// same directory. You can also find a detailed instruction on how to add
+// platforms in the `pubspec.yaml` at
+// https://flutter.dev/docs/development/packages-and-plugins/developing-packages#plugin-platforms.
+
+import 'dart:async';
+import 'dart:typed_data';
+
+import 'package:tizen_log/tizen_log.dart';
+import 'parcel.dart';
+import 'port.dart';
+import 'rpc_port_method_channel.dart';
+
+const String _logTag = 'RPC_PORT';
+
+Finalizer<StubBase> _finalizer = Finalizer<StubBase>((StubBase stub) {
+  final MethodChannelRpcPort manager = MethodChannelRpcPort.instance;
+  manager.stubDestroy(stub.portName);
+});
+
+/// The base of RpcPort stub class.
+abstract class StubBase {
+  /// The constructor of StubBase.
+  StubBase(this._portName) {
+    final MethodChannelRpcPort manager = MethodChannelRpcPort.instance;
+    manager.stubCreate(_portName);
+    _finalizer.attach(this, this);
+  }
+
+  final String _portName;
+  late final StreamSubscription<dynamic> _streamSubscription;
+  bool _isListening = false;
+
+  /// The port name of connection with proxy.
+  String get portName => _portName;
+
+  /// Sets trusted flag.
+  Future<void> setTrusted(bool trusted) async {
+    final MethodChannelRpcPort manager = MethodChannelRpcPort.instance;
+    await manager.stubSetTrusted(_portName, trusted);
+  }
+
+  /// Adds privilege required when try to connect on proxy.
+  Future<void> addPrivilege(String privilege) async {
+    final MethodChannelRpcPort manager = MethodChannelRpcPort.instance;
+    await manager.stubAddPrivilege(_portName, privilege);
+  }
+
+  /// Listens to signal of connection request from the proxy.
+  Future<void> listen() async {
```

Comment:
There's no way to stop listening for events, is this intended for TIDL?

Reply:
port.disconnect() is used when you want to disconnect the proxy connection.
TIDL will be supported disconnect method

Reply:
To stop listening for event, the stub handle of the cpp has to be released.
I think a disposal () method is needed for the user to intentionally release the handle.
If it's not provided, the class that inherited the StubBase should be nullable.

Reply:
> port.disconnect() is used when you want to disconnect the proxy connection.

I notice there are two ports (main and callback). What happens if I close just one of them? Will proxy and stub both generate disconnect event?

Reply:
The stub and the proxy get the disconnected event.